elixir/apps/web/lib/web/auth.ex

```diff
@@ -490,6 +507,22 @@ defmodule Web.Auth do
     %{}
   end
 
+  def get_client_auth_data_from_cookie(%Plug.Conn{} = conn) do
+    conn = Plug.Conn.fetch_cookies(conn, signed: [@client_auth_cookie_name])
+
+    case conn.cookies[@client_auth_cookie_name] do
+      [actor_name: _, fragment: _, identity_provider_identifier: _, state: _] = client_auth_data ->
```

Comment:
I would recommend using a map for this state because in keyword position of the keys is important, so it's not extendable. But for now keyword will work too.

Reply:
I actually wanted to a use a map here, but for some reason I kept getting a keyword list from the cookies.  I'm fairly certain I had used a map where the data was being set, but then got this when pulling the value from the cookies, so I switched to using a keyword list when setting the value, because I couldn't get the tests to pass due to what you mentioned about the position order mattering.

Let me try again with a map and see if I can reproduce what I saw.

Reply:
Oh, I see what's happening.  The `Enum.reject` line [here](https://github.com/firezone/firezone/pull/4617/files#diff-2a5232eea33367d2ec696e0907fb1198fedd504b1ca65f85b1677187b0f2a71bR133) (in the `signed_in_redirect` function) seems to be converting the Map into a Keyword List.  

I left that line as is because I figured you had it in there for a reason.  I'm guessing it's still needed, because if any of the keys had a value of `nil` then it would end up getting matched here since we're not checking for the values, but just the keys.  Any recommendations or do we just need to live with the Keyword List?

Reply:
You can use https://hexdocs.pm/elixir/main/Map.html#reject/2-examples

Reply:
👍 Updated